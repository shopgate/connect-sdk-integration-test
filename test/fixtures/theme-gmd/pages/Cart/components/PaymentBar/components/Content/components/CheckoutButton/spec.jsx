/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow } from 'enzyme';
import RippleButton from 'Components/RippleButton';
import CheckoutButton from './index';

// Mock the history connector
jest.mock('./connector', () => (obj) => {
  const newObj = obj;

  newObj.defaultProps = {
    ...newObj.defaultProps,
    isActive: false,
  };

  return newObj;
});

describe('<CheckoutButton />', () => {
  jest.useFakeTimers();

  let wrapper;
  let childButton;

  describe('Checkout is possible', () => {
    beforeEach(() => {
      wrapper = shallow(<CheckoutButton />);
      childButton = wrapper.find(RippleButton);
    });

    it('should render without any props', () => {
      expect(wrapper).toMatchSnapshot();
      expect(childButton.props().disabled).toBe(true);
    });
  });
});
