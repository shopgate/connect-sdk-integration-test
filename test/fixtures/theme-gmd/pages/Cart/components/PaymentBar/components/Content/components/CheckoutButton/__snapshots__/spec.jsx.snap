// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutButton /> Checkout is possible should render without any props 1`] = `
<RippleButton
  className=""
  disabled={true}
  flat={false}
  onClick={null}
  rippleClassName=""
  rippleSize={null}
  type="secondary"
  wrapContent={true}
>
  <Link
    className=""
    href="/checkout"
  >
    <Translate
      className=""
      params={Object {}}
      string="cart.checkout"
    />
  </Link>
</RippleButton>
`;
